pandora-common/src/assets/state/globalState.ts

```diff
@@ -172,7 +174,7 @@ export class AssetFrameworkGlobalState {
 		return new AssetFrameworkGlobalState(
 			this.assetManager,
 			newCharacters,
-			this.room,
+			newRoom,
```

Comment:
When a room state changes, all existing characters _must_ be updated using `character.updateRoomStateLink(newState, false)`, otherwise they will still have links to the old room state.
In this specific case, this would break things for multi-character room devices - when one of the characters leave, the device itself is updated, but the other character in it would still have internally link to the old state.
See `withRoomState` for the loop that needs to be added.

Reply:
You're right, I didn't notice this. I created the method `_updateRoomStateLinkOnCharacters` and added its invocation here. Additionally, I added a unit test for exactly this case. To make tests more readable I added `stateUtils.ts` as well.